ringpop.go

```diff
@@ -96,6 +96,25 @@ func mergeDefault(opts *Options) *Options {
 	return opts
 }
 
+type RingpopInterface interface {
+	Destroy()
+	Destroyed() bool
+	App() string
+	WhoAmI() string
+	Uptime() time.Duration
+	RegisterListener(l EventListener)
+	Bootstrap(opts *BootstrapOptions) ([]string, error)
+	HandleEvent(event interface{})
+	Checksum() uint32
```

Comment:
I copied all functions that were already exposed (aka starting with an upper case letter) here. But if you think it should not be in the public interface for ringpop I can remove it.

Ideally we would call this interface `Ringpop` but that might be something for a future PR because it gets kinda messy.